apis/externalsecrets/v1/secretstore_gcpsm_types.go

```diff
@@ -59,4 +59,9 @@ type GCPSMProvider struct {
 
 	// Location optionally defines a location for a secret
 	Location string `json:"location,omitempty"`
+
+	// GetLatestEnabledSecret if true, the latest enabled secret version will be fetched
+	// +optional
+	// +default=false
+	GetLatestEnabledSecret bool `json:"getLatestEnabledSecret,omitempty"`
```

Comment:
I mean this is very specific behavior, but imo (and also kubernetes API conventions), bools are an antipattern in specs. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).

Reply:
I think it makes sense, so just updated the PR accordingly with the new field you suggested 😄